core/src/test/java/org/mskcc/cbio/portal/util/TestGenePanelUtil.java

```diff
@@ -0,0 +1,242 @@
+            new HashSet<>()
+        );
+        Pair actual = getAddRemove(
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveEmptyIntersection() {
+        Pair expected = new Pair(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(piezo1, fgfr3));
+        Pair actual = getAddRemove(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(piezo1, fgfr3));
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveAOrBEmpty() {
+        Pair expected = new Pair(
+            new HashSet<>(),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1)
+        );
+        Pair actual = getAddRemove(
+            new HashSet<>(),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveIntersectAndOnlyAdd() {
+        Pair expected = new Pair(
+            makeGeneSet(otor, cadm2, msh3, p2ry10, kat2a, myb, npipb15, dtnb),
+            new HashSet<>()
+        );
+        Pair actual = getAddRemove(
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1),
+            makeGeneSet(adamts20, ablim1)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveIntersectAndOnlyRemove() {
+        Pair expected = new Pair(
+            new HashSet<>(),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, kat2a, myb, npipb15, dtnb)
+        );
+        Pair actual = getAddRemove(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testGetAddRemoveIntersectAndAndRemove() {
+        Pair expected = new Pair(
+            makeGeneSet(adamts20, ablim1),
+            makeGeneSet(piezo1, fgfr3)
+        );
+        Pair actual = getAddRemove(
+            makeGeneSet(otor, cadm2, msh3, p2ry10, adamts20, kat2a, myb, npipb15, dtnb, ablim1),
+            makeGeneSet(otor, cadm2, piezo1, msh3, fgfr3, p2ry10, kat2a, myb, npipb15, dtnb)
+        );
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testPair() {
+        Pair expected = new Pair(
+            makeGeneSet(ablim1, adamts20, fgfr3, piezo1),
+            new HashSet<>()
+        );
+        Pair actual = new Pair(
+            makeGeneSet(ablim1, fgfr3, adamts20, piezo1),
+            new HashSet<>()
+        );
+        assertEquals(expected, actual);
+    }
+    
+    private HashSet<String> makeAliases(String... aliases) {
```

Comment:
Remind me tomorrow and I'll show you a fun way to generify this and then I'll explain why its a bad idea.